module.nix

```diff
@@ -0,0 +1,111 @@
+{ config, pkgs, lib, ... }:
+let
+  cfg = config.services.arkheon;
+in
+{
+  options.services.arkheon = {
+    enable = lib.mkEnableOption "Arkheon";
+    pythonEnv = lib.mkOption {
+      internal = true;
+      visible = false;
+      type = lib.types.package;
+      default = pkgs.python3.withPackages (ps: [
+          ps.arkheon
+          ps.uvicorn
+        ]);
+      example = lib.literalExpression ''
+        pkgs.python3.withPackages (ps: [
+          ps.arkheon
+          ps.uvicorn
+          ps.gunicorn
+        ]);
+        '';
+    };
+    command = lib.mkOption {
+      internal = true;
+      visible = false;
+      type = lib.types.str;
+      description = "The command to launch the server";
+      default = "uvicorn";
+    };
+    port = lib.mkOption {
+      type = with lib.types; nullOr port;
+      default = 8007;
+      description = "Port of the server (will be passed using --bind flag)";
+    };
+    address = lib.mkOption {
+      type = lib.types.str;
+      default = "127.0.0.1";
+      description = "The address the server will listen on (will be passed using --host flag)";
+    };
+    workers = lib.mkOption {
+      type = with lib.types; nullOr ints.positive;
+      default = 4;
+      description = "number of workers";
+    };
+    flags = lib.mkOption {
+      type = with lib.types; listOf str;
+      default = [ ];
+      description = "Flags to pass to the server command-line";
+    };
+    settings = lib.mkOption {
+      type = with lib.types; submodule {
+        freeformType = attrsOf str;
+        options.SQLALCHEMY_DATABASE_URL = lib.mkOption {
+          description = "Database url";
+        };
+      };
+      description = "Settings to pass as environment variables";
+    };
+    envFile = lib.mkOption {
+      type = with lib.types; nullOr path;
+      description = "Environment file to append to settings.";
+      default = null;
+    };
+    reverseProxy = {
```

Comment:
we usually rather write `nginx` instead of `reverseProxy`
and then you plug the NGINX configuration fragment for a virtual host inside of it so you don't repeat yourself